mitodl/open-discussions
# Added user APIs

Part of #113 

#### What's this PR do?
Adds a user api and related models

#### How should this be manually tested?
- Via the DRf web interface, go to /api/v0/users/, verify you see a list of users.
- Verify you can create a new user, a username should be generated for you.
- Verify you can patch the nested profile record

#### Any background context you want to provide?
Added here is a user `Profile` model, where we store the full name and profile images. For simplicity, the profile is patching via a nested serializer.

**Note:** not dealing with any authentication/authorization in this PR, it is solely the data model and accompanying APIs.
